File: src/main/java/org/moinex/dto/TickerPerformanceDTO.java
/*
 * Filename: TickerPerformanceDTO.java
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 */

package org.moinex.dto;

import java.math.BigDecimal;

/**
 * DTO for ticker performance information
 */
public record TickerPerformanceDTO(
        String name,
        String symbol,
        BigDecimal profitLossPercentage,
        BigDecimal profitLossValue,
        BigDecimal currentValue) {

    public boolean isPositive() {
        return profitLossPercentage.compareTo(BigDecimal.ZERO) > 0;
    }

    public boolean isNegative() {
        return profitLossPercentage.compareTo(BigDecimal.ZERO) < 0;
    }

    public boolean isNeutral() {
        return profitLossPercentage.compareTo(BigDecimal.ZERO) == 0;
    }

    public String getSign() {
        if (isNeutral()) return "";
        return isPositive() ? "+ " : "- ";
    }
}

File: src/main/kotlin/org/moinex/model/dto/TickerPerformanceDTO.kt
/*
 * Filename: TickerPerformanceDTO.kt
 * Created on: January  2, 2026
 * Author: Lucas Araújo <araujolucas@dcc.ufmg.br>
 *
 * Migrate to Kotlin on 07/03/2026
 */

package org.moinex.model.dto

import java.math.BigDecimal

data class TickerPerformanceDTO(
    val name: String,
    val symbol: String,
    val profitLossPercentage: BigDecimal,
    val profitLossValue: BigDecimal,
    val currentValue: BigDecimal,
) {
    fun isPositive(): Boolean = profitLossPercentage.compareTo(BigDecimal.ZERO) > 0

    fun isNegative(): Boolean = profitLossPercentage.compareTo(BigDecimal.ZERO) < 0

    fun isNeutral(): Boolean = profitLossPercentage.compareTo(BigDecimal.ZERO) == 0

    fun getSign(): String =
        when {
            isNeutral() -> ""
            isPositive() -> "+ "
            else -> "- "
        }
}
